src/components/organisms/Chatbox/ChatForm.tsx

```diff
@@ -53,16 +52,43 @@ const ChatForm: React.FunctionComponent<PropsType> = ({
     if (!currentUserUID) return;
     switch (type) {
       case "private":
-        return (
-          discussionPartner &&
-          sendPrivateChat(currentUserUID, discussionPartner.id, text)
+        if (!discussionPartner) return;
+        return dispatch(
+          sendPrivateChat({
+            from: currentUserUID,
+            to: discussionPartner.id,
+            text,
+          })
         );
       case "global":
-        return sendGlobalChat(currentUserUID, text);
+        if (!venueId) return;
+        return dispatch(
```

Comment:
@thecadams do you know if our current babel/etc settings allow us to use the new tc39 pattern matching proposal currently? That would be the nicest solution here. For this particular use-case it would allow us to essentially 'return' a single value from the switch case, rather than needing to have individual return statements at every line:

- https://github.com/tc39/proposal-pattern-matching#motivating-examples

Reply:
It seems this is only in stage 1 of tc39 proposals, so might not be 'ready' to use yet.

Reply:
This is neat, and I'm surprised it wasn't always part of the language; I particularly like how they combine it with JSX although that'll be a win for code reduction in all `=>` functions.

As you said, I suspect it's not yet a thing we can use, looks like it hasn't progressed.

Not as pretty, and wouldn't gain much - but we could hoist the `switch`: https://stackoverflow.com/a/6612595/1013781